apps/app/src/infrastructure/storage/localstorage/AsyncStorage.ts

```diff
@@ -6,21 +6,21 @@ export const setStorage = async <T>(
   value: T,
   key: string
 ): Promise<boolean> => {
-  let response: boolean = false;
-  await storeData<T>(value, key).then((success) => {
-    !success ?? Alert.alert(error.Something_Wrong);
-    response = success;
-  });
-  return response;
+  try {
+    await storeData<T>(value, key);
+    return true;
+  } catch (e) {
+    Alert.alert(error.Something_Wrong);
+    return false;
+  }
 };
 
-export const storeData = async <T>(value: T, key: string): Promise<boolean> => {
+export const storeData = async <T>(value: T, key: string): Promise<void> => {
   try {
     const jsonValue = JSON.stringify(value);
-    await AsyncStorage.setItem(key, jsonValue);
-    return true;
+    return AsyncStorage.setItem(key, jsonValue);
   } catch (e) {
-    throw new Error(`Not able to store data for key ${key}`);
+    throw new Error(`Not able to store data for key ${key}. Error ${e}`);
```

Comment:
I was suggesting passing the cause to the second "options" argument (`new Error(message, options)`).
After looking into it a bit more, the `options` argument seems to be a pretty recent addition to javascript... so your `target` in the `tsconfig.json` has to be `ES2022` (the JS version that typescript will be compiled to). If it's set to something lower, like `ES2017`, then this feature is not available yet and you can ignore my comment.


```suggestion
    throw new Error(`Not able to store data for key ${key}`, { cause: e });
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

Reply:
I chose to include the specific support for the errors lib instead to change the overall [target](https://www.typescriptlang.org/tsconfig#target)

[updated trow error to pass cause](https://github.com/gabrielakuhn/questline-galaxy/pull/8/commits/29275b5a3074c707ece9fd7d0187ba6d332d52b3)